src/Model.ts

```diff
@@ -270,7 +259,11 @@ export default class Model extends Base {
       )
     }
 
-    const { promise, abort } = apiClient()[method](this.url(), data, otherOptions)
+    const onProgress = debounce(progress => {
+      if (optimistic && this.request) this.request.progress = progress
+    })
+
+    const { promise, abort } = apiClient()[method](this.url(), data, { onProgress, ...otherOptions })
 
     promise
```

Comment:
why are we using promises now instead of async/await?

Reply:
It's been like this since the [3.0.0 alpha PR](https://github.com/masylum/mobx-rest/pull/39/files#diff-96402e3967f40f890b345d1291b004b3R209). Do not want to introduce more entropy in this PR.